cfme/storage/volume.py

```diff
@@ -272,24 +272,29 @@ def create(self, name, storage_manager, tenant, size, provider):
             tenant: tenant name
             size: volume size in GB
             provider: provider
+            cancel: bool
 
         Returns:
             object for the :py:class: cfme.storage.volume.Volume
         """
 
         view = navigate_to(self, 'Add')
-        view.form.fill({'storage_manager': storage_manager,
-                        'tenant': tenant,
-                        'volume_name': name,
-                        'size': size})
-        view.form.add.click()
-        base_message = 'Cloud Volume "{}" created'
-        view.flash.assert_success_message(base_message.format(name))
-
-        volume = self.instantiate(name, provider)
-        wait_for(lambda: volume.exists, delay=20, timeout=500, fail_func=volume.refresh)
-
-        return volume
+
+        if not cancel:
+            view.form.fill({'storage_manager': storage_manager,
+                            'tenant': tenant,
+                            'volume_name': name,
+                            'size': size})
+            view.form.add.click()
+            base_message = 'Cloud Volume "{}" created'
+            view.flash.assert_success_message(base_message.format(name))
+
+            volume = self.instantiate(name, provider)
+            wait_for(lambda: volume.exists, delay=20, timeout=500, fail_func=volume.refresh)
+            return volume
+        else:
+            view.form.cancel.click()
+            return None
```

Comment:
`return None` is not mandatory.

Reply:
@quarckster right not mandatory but as we are returning `volume` object for `add` so better to return something for cancel well.